GVFS/GVFS.Service/GVFSService.Windows.cs

```diff
@@ -320,23 +326,22 @@ private void CreateAndConfigureProgramDataDirectories()
             Directory.SetAccessControl(serviceDataRootPath, serviceDataRootSecurity);
 
             // Special rules for the upgrader logs, as non-elevated users need to be be able to write
-            this.CreateAndConfigureUpgradeLogDirectory();
+            this.CreateAndConfigureLogDirectory(ProductUpgraderInfo.GetLogDirectoryPath());
+            this.CreateAndConfigureLogDirectory(GVFSPlatform.Instance.GetDataRootForGVFSComponent(GVFSConstants.Service.UIName));
         }
 
-        private void CreateAndConfigureUpgradeLogDirectory()
+        private void CreateAndConfigureLogDirectory(string path)

```

Comment:
Do we need to pass something to tell us what path this is?  Either the upgrader or the service or will the name of the log be enough for us to know which one?

Reply:
`CreateAndConfigureLogDirectory` is a generic method. It doesn't need to know what process/function the log-directory is associated with.